### DISCUSSION TITLE
Copying image using imagemagick, file size goes up?

### DISCUSSION THREAD

**Discussion by Jakobohrnell** [ImageMagick/ImageMagick#5592] [Help]
Hello, I've got a peculiar set of images which when processed, in any way at all, go up in size. Could someone help me figure out why?
I'm trying with the following command
`magick.exe .\original.png -strip -quality 100 .\copy.png`
but even a simple
`magick.exe .\original.png .\copy.png`
makes it increase in size.

Running the original file through optipng or oxipng yields nothing, it simply says it's already optimized. Running the copy through them produces a slight decrease in size, but it's still not close to the size of the original.

The increase is very slight but I'm looking to store very large amounts of processed images, so it is of importance. I'm attaching the images in question. According to imagemagick, the signature and the compression method of both says "zip".

COPY
![copy](https://user-images.githubusercontent.com/26467980/192153876-3e0714a2-c05e-4428-892b-676e8a42cfe6.png)

ORIGINAL
![original](https://user-images.githubusercontent.com/26467980/192153880-9df8aff9-21f3-40f5-9fb4-9d9714ed2bce.png)

**Comment by fmw42** [comment]
PNG quality is not the same scale as JPG quality. It is not 0 to 100 %.  See https://imagemagick.org/script/command-line-options.php#quality.  Also your input may have been highly optimized and the output may not be the same compression.

**Reply by Jakobohrnell** [reply]
Surely oxipng or optipng would've been able to achieve the same compression? They cycle through all png compression options and find the best one for the file.

**Comment by urban-warrior** [comment]
Try this command:

```
magick -define PNG:compression-level=9 -define png:compression-strategy=1 -define png:compression-filter=2 original.png copy.png
```

It returns a copy of the original slightly smaller in size.

**Reply by Jakobohrnell** [reply]
The output of that command is bigger than the original for me.

**Comment by urban-warrior** [comment]
We're using ImageMagick 7.1.0-49 and libpng 1.6.37:

```
$ magick -define PNG:compression-level=9 -define png:compression-strategy=1 -define png:compression-filter=2 original.png copy.png
$ ls -l *png
-rz-------. 1 lando 1350355 Sep 25 14:59 copy.png
-rz-------. 1 lando 1369194 Sep 25 14:45 original.png
```

Try instead, `-quality 97`.

**Reply by Jakobohrnell** [reply]
The original should be 1226836 bytes, you've used that command on the copy. Your command, used on the original, still causes an increase for me. I'm on windows, if that matters.

**Comment by urban-warrior** [comment]
I downloaded `orginal.png` from this discussion.  It was 1369194 bytes.   To get optimal compression, you could try `pngcrush`.  Although it created a file larger than what ImageMagick returned @ 1368007 bytes.

**Reply by Jakobohrnell** [reply]
hmmm, could you please redownload and tell me the byte size of both files for you? Because for me the original is 1 226 836 and the copy is 1 369 194.

**Comment by urban-warrior** [comment]
`pngcheck` shows a bunch of IDAT chunks.  We increased the compression buffer size, which reduces the IDAT chunks but the result was still bigger than the original image.  You may need to use `oxipng` to compress the image.  It returned a copy of the image about 1000 bytes smaller than the original.
